game.cpp

```diff
@@ -1216,39 +1241,41 @@ void test_game() //!OCLINT tests may be many
   }
 #endif
 
+#define FIX_ISSUE_238
 #ifdef FIX_ISSUE_238
-  //Food and player can be overlapped
+  // The game can be checked for any collision between food and players
   {
     game g;
-    assert(!has_player_food_collision(g));
+    assert(!any_player_food_collision(g));
     put_player_on_food(g.get_player(0), g.get_food()[0]);
-    assert(has_player_food_collision(g));
+    assert(any_player_food_collision(g));
   }
 #endif
 
+#define FIX_ISSUE_244
 #ifdef FIX_ISSUE_244
-
   {
     game g;
     const auto init_player_size = get_nth_player_size(g,0);
     put_player_on_food(g.get_player(0), g.get_food()[0]);
     g.tick();
     assert(g.get_player(0).get_diameter() > init_player_size);
-
   }
 
 #endif
 
+#define FIX_ISSUE_247
 #ifdef FIX_ISSUE_247
   {
-    player p;
-    food f;
-    assert(!player_and_food_are_colliding(p,f));
+    double p_x = 12.3;
+    double f_x = p_x + 1.0;
+    player p{p_x};
+    food f{f_x};
+    assert(!in_contact_with_uneaten_food(p,f));
```

Comment:
I think these lines are a bit unnecessary and make the test read less smoothly, while not conveying more meaning. I would revert back to the initial version.

Reply:
I did this bc players and food items had the same default position of (0, 0) (and since food is uneaten, they were colliding).
Though this is no longer true, I think it's clearer to explicitly set them to different position. But I can collapse 1270-1273 into two lines.